driver-core/src/main/com/mongodb/internal/capi/MongoCryptHelper.java

```diff
@@ -55,16 +56,21 @@ public static MongoCryptOptions createMongoCryptOptions(final Map<String, Map<St
                                 .build()
                 );
             } else {
-                throw new MongoClientException("Unrecognized KMS provider key: " + entry.getKey());
+                bsonKmsProviders.put(entry.getKey(), new BsonDocumentWrapper<>(new Document(entry.getValue()), new DocumentCodec()));
             }
         }
+
+        if (!bsonKmsProviders.isEmpty()) {
```

Comment:
Should it be an error if there are no providers?  Or is that already detected at a level up?

Reply:
Handled by MongoCryptOptions